### PR TITLE
RavenDB-3810: Add "Received" log line when request arrives. The requests

### PR DESCRIPTION
**Pull request by aviviadi**
numbering is being set on arrival and not on action finalizing.

### CODE REVIEW

File: Raven.Database/Server/WebApi/RequestManager.cs [hunk 1]
```diff
@@ -269,7 +271,57 @@ public void Dispose()
 			return response;
 		}
 
-		/// <summary>
+	    private long PreActionLog(IResourceApiController controller)
+	    {
+            LogHttpRequestStatsParams logHttpRequestStatsParam = null;
```
Comment (ayende) [inline Raven.Database/Server/WebApi/RequestManager.cs]: You should only do all this work here if and only if you have log.IsDebugEnabled

File: Raven.Database/Server/WebApi/RequestManager.cs [hunk 2]
```diff
@@ -269,7 +271,57 @@ public void Dispose()
...
+            try
+            {
+                StringBuilder sb = null;
+                if (controller.CustomRequestTraceInfo != null && controller.CustomRequestTraceInfo.Count > 0)
```
Comment (ayende) [inline Raven.Database/Server/WebApi/RequestManager.cs]: Controller can't have customer request trace info before the request.

File: Raven.Database/Server/WebApi/RequestManager.cs [hunk 3]
```diff
@@ -269,7 +271,57 @@ public void Dispose()
...
+                        if (!char.IsWhiteSpace(sb[sb.Length - 1]))
+                            break;
+                        sb.Length--;
+                    }
+                }
+                var innerRequest = controller.InnerRequest;
+                var httpRequestHeaders = innerRequest.Headers;
+                var httpContentHeaders = innerRequest.Content == null ? null : innerRequest.Content.Headers;
+                Stopwatch dummyStopwatch = new Stopwatch();
+                logHttpRequestStatsParam = new LogHttpRequestStatsParams(
```
Comment (ayende) [inline Raven.Database/Server/WebApi/RequestManager.cs]: We haven't done anything to the request yet, we just need the URL / database, nothing more.

**Comment by ayende** [comment]
Please squash your changes.

**Comment by aviviadi** [comment]
This PR closed.  Instead use : https://github.com/ayende/ravendb/pull/2720

**Pull request closed without merging**